privaterelay/settings.py

```diff
@@ -519,6 +519,19 @@ def _get_initial_middleware():
 
 WHITENOISE_INDEX_FILE = True
 
+
+# See
+# https://whitenoise.evans.io/en/stable/django.html#WHITENOISE_ADD_HEADERS_FUNCTION
+# Intended to ensure that the homepage does not get cached in our CDN,
+# so that the `RedirectRootIfLoggedIn` middleware can kick in for logged-in
+# users.
+def set_index_cache_control_headers(headers, path, url):
+    if path == os.path.abspath("./frontend/out/index.html"):
```

Comment:
This works for `DEBUG=True` but fails in production mode. Here's a version that works in both:

```python
    if DEBUG:
        home_path = os.path.join(BASE_DIR, "frontend/out", "index.html")
    else:
        home_path = os.path.join(STATIC_ROOT, "index.html")

    if path == home_path:
        headers["Cache-Control"] = "no-cache, public"
```

Reply:
They are different folders:

```
STATIC_ROOT = os.path.join(BASE_DIR, "staticfiles")
STATICFILES_DIRS = [
    os.path.join(BASE_DIR, "frontend/out"),
    os.path.join(BASE_DIR, "static"),
]
```

In `DEBUG=True`, the file is read from the source at `[proj]/frontend/out/index.html`. When building for prod with `manage.py collectstatic`, they are collected from the `STATICFILES` directories into `STATIC_ROOT`, so it is at `[proj]/staticfiles/index.html`.

15 years ago, we would have collected them to `/var/www/fx-private-relay/static` and served with Apache. 10 years ago, `manage.py collectstatic` would have uploaded them to an S3 bucket, and `STATIC_URL` would be on the CDN domain. Let's see what we're doing in 2030...

Reply:
Gotcha, cheers!